### PR TITLE
comcast.net - Use OAuth2

### PR DESCRIPTION
**Pull request by abrotman**
Update documentation, add OAuth2 stanzas

### CODE REVIEW

File: ispdb/comcast.net.xml [hunk 1]
```diff
@@ -9,46 +9,61 @@
       <port>993</port>
       <socketType>SSL</socketType>
       <username>%EMAILADDRESS%</username>
+      <authentication>OAuth2</authentication>
```
Comment (benbucksch) [inline ispdb/comcast.net.xml:12]: Hey Alex, have you tested how a Thunderbird _without_ the OAuth info hardcoded (and without the ability to use the OAuth config provided here) will react to this? I don't remember whether TB skips OAuth configs, if TB doesn't know the OAuth config, and falls back to password auth as before (that's what it should do), or whether it tries to use OAuth2 and then fails during login (that's how I read the code - this should be fixed first).

File: ispdb/comcast.net.xml [hunk 2]
```diff
@@ -9,46 +9,61 @@
...
       <descr lang="es">Instrucciones para instalar Thunderbird para el uso de Comcast para dueños de casa.</descr>
     </documentation>
+    <documentation url="https://www.xfinity.com/support/articles/third-party-email-access">
+      <descr lang="en">Instructions for enabling IMAP/POP/SMTP access for third-party clients</descr>
+    </documentation>
   </emailProvider>
+  <oAuth2>
+    <issuer>comcast.net</issuer>
+    <scope>https://email.comcast.net/ profile openid</scope>
+    <authURL>https://oauth.xfinity.com/oauth/authorize</authURL>
```
Comment (benbucksch) [inline ispdb/comcast.net.xml:66]: I've checked WHOIS and DNS, and xfinity.com is owned by comcast.net.

File: ispdb/comcast.net.xml [hunk 3]
```diff
@@ -9,46 +9,61 @@
...
       <descr lang="en">Instructions for setting up Thunderbird for use with comcast.net for home users.</descr>
       <descr lang="es">Instrucciones para instalar Thunderbird para el uso de Comcast para dueños de casa.</descr>
     </documentation>
+    <documentation url="https://www.xfinity.com/support/articles/third-party-email-access">
+      <descr lang="en">Instructions for enabling IMAP/POP/SMTP access for third-party clients</descr>
+    </documentation>
   </emailProvider>
+  <oAuth2>
+    <issuer>comcast.net</issuer>
+    <scope>https://email.comcast.net/ profile openid</scope>
```
Comment (benbucksch) [inline ispdb/comcast.net.xml:65]: Do we need the `openid` scope for IMAP, POP3 and SMTP? Do we need `profile`?

**Comment by benbucksch** [comment]
I've checked WHOIS and DNS, and xfinity.com is owned by comcast.net.

File: ispdb/comcast.net.xml (hunk 3 above)
Comment (abrotman) [reply ispdb/comcast.net.xml:65]: This is what they told me, so that's what I've done.

File: ispdb/comcast.net.xml (hunk 1 above)
Comment (abrotman) [reply ispdb/comcast.net.xml:12]: I don't believe I have a reasonable way to test this?  However, Tbird is not the only consumer of this.  FWIW, I'm also about submit a patch to Tbird to enable/hardcode Oauth for the same set of users.

Comment (benbucksch) [reply ispdb/comcast.net.xml:12]: You can try with Thunderbird 102. To use your config file, you can place it on your disk in the TB install directory in file _tb-install-dir_/isp/comcast.net.xml . (Details on https://www.bucksch.org/1/projects/thunderbird/autoconfiguration/ )

Comment (abrotman) [reply ispdb/comcast.net.xml:12]: (https://bugzilla.mozilla.org/show_bug.cgi?id=1844810)

Comment (benbucksch) [reply ispdb/comcast.net.xml:12]: I tested it, and unfortunately, it fails :-( I'll post another message on the PR, outside the review here.

**Comment by benbucksch** [comment]
If I take this config in a TB 102, without patches, it fails. It accepts the OAuth2 auth method, but then doesn't have the OAuth2 server config hardcoded, and also isn't able to use the OAuth2 server config from the autoconfig file here, therefore failing the authentication.

That's a TB bug, we need to fix that, before we can deploy this config. I'm sorry for that :-( What should be happening is that the client ignores any config that it cannot _actually_ use.
 
image

**Comment by abrotman** [comment]
There's a patch waiting for Thunderbird which would resolve that issue.  I assume once that is accepted/deployed, this becomes a non-issue.

**Comment by benbucksch** [comment]
Yes, once it's deployed to users.

File: ispdb/comcast.net.xml (hunk 3 above)
Comment (benbucksch) [reply ispdb/comcast.net.xml:65]: Yes, we now know from testing that we need those.

**Review by benbucksch — APPROVED** [review]
The change looks good to me. Therefore, r+

However, we CANNOT deploy this yet, because Thunderbird versions without the hardcoded client ID will fail. We need to wait until these TB versions have less than 2-5% of usage in the field.

**Comment by benbucksch** [comment]
Do NOT merge until either [Bug 184481](https://bugzilla.mozilla.org/show_bug.cgi?id=1844810) or [1869122](https://bugzilla.mozilla.org/show_bug.cgi?id=1869122) in Thunderbird have deployed *and* have reached 95-98% of the actual userbase.

**Comment by abrotman** [comment]
Is this ready to merge?

**Comment by Betterbird** [comment]
You should backport https://bugzilla.mozilla.org/show_bug.cgi?id=1869122 to TB 115.x.

**Review by cketti — APPROVED** [review]
The OAuth credentials for Comcast are included in Thunderbird 115.6.1+. That's sufficiently widespread by now.

**Comment by cketti** [comment]
Thanks @abrotman :+1:

**Pull request merged**